### ISSUE TITLE
Combine GPS and distance to dected objects to improve localization

### ISSUE THREAD

**Issue by Lukasnol** [una-auxme/paf#458]
### Detailed Description

Use the distance to recognised objects in combination with the data measured by the GPS to better estimate your own position and thus reduce noise.
First make sure that the distance calculation to recognised objects works well enough.

### Definition of Done

- see description above

**Comment by ll7** [comment]
In my opinion, this issue appears to be very difficult. If I understand you correctly, you would like to use dynamic objects for your localization?

Before you use this approach, I would recommend spending at least a small amout of time looking in to point cloud localization methods. If I am correct, high definition maps of the static buildings should be available. You should be able to use these buildings as reference point. However, filtering #459 would be my first approach.

These are only my thoughts, you can do whatever you like.

**Comment by Toni2go** [comment]
Yes, we also thought that this issue might be quite difficult and take a lot of time so we are planning on solving the other issues first.
To improve the localization (in particular the estimation of the position) we were planning on using static objects like traffic lights or speed limit signs but buildings as reference points seem to be a good solution as well.

**Comment by Johannes1098** [comment]
Extendend Kalman Filter was implemented in PAF24. 
See https://github.com/una-auxme/paf/pull/693
Improved our localization so this issue was not pursued further
